src/cardstack/al-cardstack-view.ts

```diff
@@ -285,41 +323,82 @@ export abstract class AlCardstackView< EntityType=any,
                 caption: properties.caption
             };
         } );
-        this.cards.push( ...newData );
-        this.cards = this.cards.map( c => this.evaluateCardState( c ) );
-        this.visibleCards = this.cards.reduce( ( count, card ) => count + ( card.visible ? 1 : 0 ), 0 );
-
-        if (this.localPagination) {
-            this.markCardsAsCheck();
-        }
-        return this.visibleCards;
+        return newData;
     }
 
     /**
      *  Method to determine visibility of an individual card item based on the current set of active filters.
      */
-    protected evaluateCardState( card:AlCardstackItem<EntityType,PropertyType> ):AlCardstackItem<EntityType,PropertyType> {
-        card.visible = true;
+    protected evaluateCardVisibilityByFilter( card:AlCardstackItem<EntityType,PropertyType> ):boolean {
+        let visible = true;
         let filterProperties = Object.keys( this.activeFilters );
         if ( filterProperties.length === 0 ) {
-            return card;
+            return true;
         }
-        filterProperties.find( property => {
+        filterProperties.forEach( property => {
```

Comment:
I think this should still use `find` to terminate as soon as a card fails to match a filter.